File: frontend/types/post.ts
export type PostStatus = 'DRAFT' | 'PUBLISHED';

export interface Post {
  id: number;
  title: string;
  slug: string;
  excerpt: string;
  content: string;
  coverImageUrl?: string;
  status: PostStatus;
  authorName: string;
  categoryName?: string;
  tags: string[];
  createdAt: string;
  updatedAt: string;
  publishedAt?: string;
}

export interface PostRequest {
  title: string;
  content: string;
  excerpt?: string;
  coverImageUrl?: string;
  status: PostStatus;
  categoryId?: number;
  tagIds?: number[];
}

File: backend/src/main/java/com/blog/cms/model/Post.java
package com.blog.cms.model;

import lombok.*;
import org.springframework.data.annotation.Id;
import org.springframework.data.relational.core.mapping.Table;

import java.time.LocalDateTime;

@Table("posts")
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Post {

    @Id
    private Long id;
    private String title;
    private String slug;
    private String excerpt;
    private String content;
    private String coverImageUrl;
    private String status;       // DRAFT | PUBLISHED
    private Long authorId;
    private Long categoryId;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
    private LocalDateTime publishedAt;
}
